Dockerfile

```diff
@@ -1,48 +1,41 @@
 # Build: sudo docker build -t <project_name> .
 # Run: sudo docker run -v $(pwd):/workspace/project --gpus all -it --rm <project_name>
 
+ARG PYTORCH="1.11.0"
+ARG CUDA="11.5"
+ARG CUDNN="8"
+ARG UBUNTU="22.04"
 
-FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04
-
-
-ENV CONDA_ENV_NAME=myenv
-ENV PYTHON_VERSION=3.8
-
+FROM nvcr.io/nvidia/pytorch:22.05-py3
 
 # Basic setup
 RUN apt update
-RUN apt install -y bash \
-    build-essential \
-    git \
-    curl \
-    ca-certificates \
-    wget \
-    && rm -rf /var/lib/apt/lists
-
 
 # Set working directory
 WORKDIR /workspace/project
 
-
 # Install Miniconda and create main env
-ADD https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh miniconda3.sh
-RUN /bin/bash miniconda3.sh -b -p /conda \
-    && echo export PATH=/conda/bin:$PATH >> .bashrc \
-    && rm miniconda3.sh
-ENV PATH="/conda/bin:${PATH}"
-RUN conda create -n ${CONDA_ENV_NAME} python=${PYTHON_VERSION}
-
+# ADD https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh miniconda3.sh
+# RUN /bin/bash miniconda3.sh -b -p /conda \
+#     && echo export PATH=/conda/bin:$PATH >> .bashrc \
+#     && rm miniconda3.sh
+# ENV PATH="/conda/bin:${PATH}"
+# RUN conda create -n ${CONDA_ENV_NAME} python=${PYTHON_VERSION}
 
 # Switch to bash shell
-SHELL ["/bin/bash", "-c"]
-
+# SHELL ["/bin/bash", "-c"]
 
 # Install requirements
 COPY requirements.txt ./
```

Comment:
Now that I think about it, it would probably be better to just specify all the dependencies explicitly in this file instead of relying on copying requirements

Reply:
> Now that I think about it, it would probably be better to just specify all the dependencies explicitly in this file instead of relying on copying requirements

I proposed this docker because the current one is outdated. And I've read a lot about the nvidia docker, and it turns out that they perform internal optimizations more than just installing packages.

Feel free to change it.

Reply:
@ashleve Copying `requirements.txt` seems more flexible tho, because if you change the file you have to remember to update it in Dockerfile. Which seems like a decent way to footgun yourself

Reply:
hi guys